Scripts/static-analyzer/Sources/Analyzer/USR.swift

```diff
@@ -0,0 +1,17 @@
+import Foundation
+
+/// A type represent "Unified Symbol Resolution" string.
+///
+/// From [this random API doc in clang](https://libclang.readthedocs.io/en/latest/#clang.cindex.Cursor.get_usr):
+/// > A Unified Symbol Resolution (USR) is a string that identifies a particular entity (function, class, variable, etc.) within a program.
```

Comment:
Additional doc I found googling:

> A symbol can not be identified by its spelling only: context information is needed to disambiguate uses of the same spelling in different scopes. In order to uniquely identify a symbol across all translation units in a project, clang defines Unified Symbol Resolutions. For example the USR of the display symbol referred to on line 30 is `c:MyClass>#ddisplay#` whereas line 33 refers to the symbol with USR `c:MyClass>#Idisplay#`
>
> https://ffevotte.github.io/clang-tags/

The lines referenced in the example are

```
26: int main () {
27:   int display = 3;                                       //
28: 
29:   MyClass<double> a;                                     //
30:   a.display();                                           //
31: 
32:   MyClass<int> b;
33:   b.display();                                           //
34: }
```